[Ng Yan Jia] iP

sprintaway opened the pull request:
# **Duke CLI Chatbot**
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Duke CLI Chatbot frees your mind of having to remember things you need to do. It's,
- text-based
- easy to use
- ~~FAST~~ **__SUPER FAST__** to use

All you need to do is,
1. download it from [here](https://github.com/sprintaway/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

#### If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

sprintaway committed:
Implemented Level-1

sprintaway committed:
Implement Level-2

sprintaway committed:
Implemented Level-3

sprintaway committed:
Implemented Level-4

sprintaway committed:
Implemented A-TextUiTesting

sprintaway committed:
Implemented Level-5

sprintaway committed:
Implemented Level-6

sprintaway committed:
Implement Level-7

sprintaway committed:
Implement Level-8

sprintaway committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

sprintaway committed:
Edited merging issues

sprintaway committed:
Comit

sprintaway committed:
wwef

sprintaway committed:
Edit duke.txt to make it readable

sprintaway committed:
Make the code more OOP

sprintaway committed:
Add packages

sprintaway committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

sprintaway committed:
Add JUnit tests

sprintaway committed:
Removed emoji intefering with gradle, added javadocs

sprintaway committed:
Improve coding standard

sprintaway committed:
Ensure coding standard

sprintaway committed:
Implement Level-9

sprintaway committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/duke/Duke.java
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/task/Task.java

sprintaway committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/parser/Parser.java

sprintaway committed:
Remove error messages

sprintaway committed:
Add missing javadocs

sprintaway committed:
Add checkstyle check and fix code to adhere to checkstyle

sprintaway committed:
Add Level-10

sprintaway committed:
Make minor CSS changes

parth-io commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,229 @@
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+import javafx.util.Duration;
```
This import statement can be brought above, to make the grouping more consistent

parth-io commented on src/main/java/duke/Launcher.java:
```diff
@@ -0,0 +1,12 @@
+    public static void main(String[] args) {
+        Application.launch(Main.class, args);
+    }
+}
```
Would be good to add a newline here, for consistency

parth-io commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,28 @@
+     *
+     * @return true if the command ends the program and vice versa
+     */
+    public abstract boolean isExit();
```
Minor issue, the name for this method sounds like it's been named for a boolean variable. Something like `hasExit()` might sound better

parth-io commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,135 @@
+        }
+        return null;
+    }
+}
```
This file is very easy to read, despite the long methods, so kudos to you!

parth-io commented on src/test/java/duke/task/DeadlineTest.java:
```diff
@@ -0,0 +1,22 @@
+
+public class DeadlineTest {
+    @Test
+    public void addToDoTest() {
```
Minor naming issue, but maybe the test method name can be simplified to `addToDo()`, as the JUnit annonation does suggest it's a test method. Just reads more cleanly

parth-io reviewed (commented):
LGTM, only minor issues

Vinodjayakumar124 commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,127 @@
+     *
+     * @return the size of the taskList
+     */
+    public int taskListSize() {
```
Perhaps replacing this method name with a getSize may make this clearer to other developers

Vinodjayakumar124 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,50 @@
+     *
+     * @return string containing task to be done
+     */
+    private String getSubstring() {
```
You could also consider renaming this method to something like getTaskDescription so as to understand its purpose without reading the docstring

Vinodjayakumar124 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,160 @@
+     * @param taskList the tasklist to be displayed
+     * @return string that contains tasks found
+     */
+    public String showTaskFound(TaskList taskList) {
```
Since there might be more than 1 task that are matched, perhaps we could rename this method to showTasksFound

Vinodjayakumar124 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,160 @@
+     */
+    public String showTaskAdded(TaskList taskList) {
+        return ("Got it. I've added this task:")
+                + "\n" + (taskList.latestTask() + "\nNow you have " + taskList.taskListSize()
```
I really liked there is a consistent breaking up of long lines into two or more lines across the entire project

Vinodjayakumar124 reviewed (commented):
LGTM. I really liked the structure of the entire project and your code was really easy to read and understand even without the javadoc so good job. Commented on a couple of nits that I found.

sprintaway committed:
Add assertions to the program

Assertions are important as it can verify certain
assumptions of a program that must hold true at a given point of time.

Let's
* add various assertions to commands to ensure valid commands
* check if storage path is not empty
* assert conditions in parser that must hold true

sprintaway committed:
Delete dead code and added suggestions made by PR reviewers, and
attempt to resolve deep nesting in storage class. Not very
successful as one needs to choose between sacrificing code
repetition or deep nesting.

This is needed to improve code quality and to enhance readability of
code.

Attempt to avoid complicated expressions in parser result in
redeclaring of several local variables and code duplication,
as well as deep nesting.

Therefore parser remains as it is.

sprintaway committed:
Merge pull request #1 from sprintaway/branch-A-Assertions

Add assertions to the program

Assertions are important as it can verify certain
assumptions of a program that must hold true at a given point of time.

Let's

add various assertions to commands to ensure valid commands
check if storage path is not empty
assert conditions in parser that must hold true

sprintaway committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	data/duke.txt

sprintaway committed:
Merge pull request #2 from sprintaway/branch-A-CodeQuality

Delete dead code and added suggestions made by PR reviewers, and

Attempt to resolve deep nesting in storage class. Not very
successful as one needs to choose between sacrificing code
repetition or deep nesting.

This is needed to improve code quality and to enhance readability of
code.

Attempt to avoid complicated expressions in parser result in
redeclaring of several local variables and code duplication,
as well as deep nesting.

Therefore parser remains as it is.

sprintaway committed:
Add gradle.yml to set up CI

sprintaway committed:
Make the GUI work again

sprintaway committed:
The program lacks a way to sort by dates chronologically.

This is not ideal as users want to see which tasks are due sooner
than others.

Let's
* create a sort command that sorts by date chronologically
* todo should be at the least priority since it lacks a date
* the list is from most urgent to least urgent

sprintaway committed:
Reduce method size of parse() through reformat of code
Check checkstyle through gradle
Remove unused code
Add javadocs
Edit header comments

sprintaway committed:
Fix wrong ordering of sort
Add Ui.png
Update readme

sprintaway committed:
Fix bug caused by .jpg file in linux to not display,
use .png file instead to try

sprintaway committed:
Remove * for import in Parser class

sprintaway committed:
Test if it now works properly on linux
